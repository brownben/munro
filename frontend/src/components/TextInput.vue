<template>
  <div class="text-input top--4 relative">
    <label
      class="font-heading text-sm text-main p-1 mx-3 bg-white relative select-none bottom--3 pb-0"
      >{{ label }}</label
    >
    <input
      :value="value"
      :type="type"
      class="w-full border border-main rounded-tl-lg rounded-br-lg px-4 py-2 text-body outline-none"
      @input="$emit('input', $event.target.value)"
    />
  </div>
</template>

<script>
export default {
  name: 'TextInput',

  props: {
    value: {
      type: String,
      default: '',
    },
    label: {
      type: String,
      default: '',
    },
    type: {
      type: String,
      default: 'text',
    },
  },
}
</script>
<style>
.top--4 {
  top: -1rem;
}

.bottom--3 {
  bottom: -0.75rem;
}
</style>
