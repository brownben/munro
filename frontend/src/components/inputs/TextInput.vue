<template>
  <div>
    <label
      class="block pb-1 select-none font-heading"
      :class="focus ? 'text-main-600' : 'text-gray-600'"
      :for="label"
    >
      {{ label }}
    </label>
    <input
      :id="label"
      :value="value"
      :type="type"
      class="w-full px-3 py-2 font-sans text-gray-900 transition duration-300 ease-in-out bg-white border outline-none appearance-none rounded-shape focus:shadow-outline focus:border-main-400"
      @input="$emit('input', $event.target.value)"
      @focus="focus = true"
      @blur="focus = false"
    />
  </div>
</template>

<script>
export default {
  name: 'TextInput',

  props: {
    value: {
      type: String,
      default: '',
    },
    label: {
      type: String,
      default: '',
    },
    type: {
      type: String,
      default: 'text',
    },
  },

  data: () => ({
    focus: false,
  }),
}
</script>
