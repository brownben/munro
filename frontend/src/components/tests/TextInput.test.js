import { mount } from '@vue/test-utils'
import TextInput from '@/components/TextInput'

test('Is a Vue Instance', () => {
  const wrapper = mount(TextInput)
  expect(wrapper.isVueInstance()).toBeTruthy()
})

test('Renders Correctly', () => {
  const wrapper = mount(TextInput, {
    propsData: {
      value: 'hello',
      label: 'Test',
      type: 'url',
    },
  })
  expect(wrapper.element).toMatchSnapshot()
})
