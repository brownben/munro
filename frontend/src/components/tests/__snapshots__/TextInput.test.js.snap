// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<div
  class="text-input top--4 relative"
>
  <label
    class="font-heading text-sm text-main p-1 mx-3 bg-white relative select-none bottom--3 pb-0"
  >
    Test
  </label>
   
  <input
    class="w-full border border-main rounded-tl-lg rounded-br-lg px-4 py-2 text-body outline-none"
    type="url"
  />
</div>
`;
